data_generation/utils/chatgpt_prompt_utils.py

```diff
@@ -0,0 +1,91 @@
+import ast
+
+import openai
+from openai.error import ServiceUnavailableError
+
+
+def get_chatgpt_response(content, model="gpt-3.5-turbo"):
+
+    completion = openai.ChatCompletion.create(
+        model=model,
+        messages=[
+            {
+                "role": "user",
+                "content": content,
+            }
+        ],
+    )
+    gpt_response = completion.choices[0].message.content
+    return gpt_response
+
+
+def get_exam_questions_from_chatgpt(course: str, chapter: str, topic: str):
+
+    prompt = (
+        f"I wanna generate mock exam question for {course=}, {chapter=}, {topic=}. "
+        "Come up with a multiple-choice questions for the subtopic.\n"
+        "We are using these questions to test people's understanding of the subtopic,\n"  # test this
+        "so focus more on testing peoples understanding. \n"  # test this if necessary
+        "On a scale of 1 to 10 make all the questions difficulty between 5 and 10 with increasing difficulty.\n"
+        "Make sure to make each question different from each other.\n"
+        "an example question is as follows\n"
+        "Which of the following best describes the Binary Search algorithm?\n"
+        "a) A linear search algorithm with a complexity of O(n).\n"
+        "b) A divide-and-conquer algorithm that works on a sorted array with a complexity of O(log n).\n"
+        "c) An algorithm that only works on doubly linked lists with a complexity of O(n^2).\n"
+        "d) An iterative algorithm that performs a breadth-first search on a tree with a complexity of O(n).\n"
+        "make sure to put `---` between each question. Don't give the answer of the question"
+    )
+    print(prompt)
```

Comment:
printing should be removed. They are for debugging. Alternatively, you can add logging: https://docs.python.org/3/library/logging.html

See much easier logging library: https://github.com/Delgan/loguru